### PR TITLE
HttpStress: Add scripts for loading the corefx testhost in the environment

### PR DESCRIPTION
**Pull request by eiriktsarpalis**
Adds a pair of bash & powershell scripts intended for loading the `testhost` sdk in the current environment. Effectively it can be used to run httpstress against the current corefx in three commands:
```powershell
# Build corefx from source
PS> .\build.sh -c Release
# Load the testhost sdk in the current environment, must match build configuration
PS> . .\src\System.Net.Http\tests\StressTests\HttpStress\load-corefx-testhost.ps1 -c Release
# run the stress suite with the new bits
PS> cd .\src\System.Net.Http\tests\StressTests\HttpStress ; dotnet run -r win10-x64 
```
Equivalently using bash
```bash
# Build corefx from source
$ ./build.sh -c Release
# Load the testhost sdk in the current environment, must match build configuration
$ source src/System.Net.Http/tests/StressTests/HttpStress/load-corefx-testhost.sh -c Release
# run the stress suite with the new bits
$ cd src/System.Net.Http/tests/StressTests/HttpStress && dotnet run -r linux-x64 
```

The powershell script has been tested with powershell core on linux and the bash script works with mingw on Windows. I've placed the scripts in the stress folder for now, but I think there's value in moving these more centrally in the repo.

Motivation is simplifying steps required for running HttpStress in CI.

### CODE REVIEW

File: src/System.Net.Http/tests/StressTests/HttpStress/load-corefx-testhost.ps1 [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    $pathSeparator=if($os -eq "Windows_NT") { ";" } else { ":" }
+    
+    $env:DOTNET_ROOT=$candidate_path
+    $env:DOTNET_CLI_HOME=$candidate_path
+    $env:PATH=($candidate_path + $pathSeparator + $env:PATH)
+    $env:DOTNET_MULTILEVEL_LOOKUP=0
+    $env:DOTNET_ROLL_FORWARD_ON_NO_CANDIDATE_FX=2
+}
+
+applyToEnvironment
```
Comment (davidsh) [inline src/System.Net.Http/tests/StressTests/HttpStress/load-corefx-testhost.ps1]: nit: missing end-of-line character

File: src/System.Net.Http/tests/StressTests/HttpStress/load-corefx-testhost.sh [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+    if which cygpath > /dev/null 2>&1; then
+        # cygwin & mingw compat: PATH values must be unix style
+        export PATH=$(cygpath -u $candidate_path):$PATH
+    else
+        export PATH=$candidate_path:$PATH
+    fi
+}
+
+apply_to_environment
```
Comment (davidsh) [inline src/System.Net.Http/tests/StressTests/HttpStress/load-corefx-testhost.sh]: nit: missing end-of-line character

File: src/System.Net.Http/tests/StressTests/HttpStress/Program.cs [hunk 3]
```diff
@@ -217,9 +217,8 @@ private static async Task WaitUntilMaxExecutionTimeElapsedOrKeyboardInterrupt(Ti
         return value is null ? null : new S?(mapper(value.Value));
     }
 
-    private static string GetSysNetHttpAssemblyInfo()
+    private static string GetAssemblyInfo(System.Reflection.Assembly assembly)
```
Comment (scalablecory) [inline src/System.Net.Http/tests/StressTests/HttpStress/Program.cs]: Looks like this can be a local function?

**Review by scalablecory — APPROVED** [review]
lgtm, though I'm no expert in bash.

**Review by davidsh — APPROVED** [review]

**Review by ViktorHofer — APPROVED** [review]
You might want to follow the powershell function naming guidelines - or not, depends what's more important to you, cross plat consistency or ecosystem familiarity.

It's a bit unfortunate that you need to set a roll forward policy but I don't know if a good way to the latest tfm from outside msbuild. We could call into it but I think this is fine.

**Review by ViktorHofer — APPROVED** [review]

**Pull request merged**